x-pack/plugins/rollup/server/lib/search_strategies/rollup_search_strategy.ts

```diff
@@ -4,44 +4,54 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import { keyBy, isString } from 'lodash';
-import { KibanaRequest } from 'src/core/server';
 
-import { CallWithRequestFactoryShim } from '../../types';
+import { ReqFacade } from '../../../../../../src/plugins/vis_type_timeseries/server';
+import { ENHANCED_ES_SEARCH_STRATEGY } from '../../../../data_enhanced/server';
 import { mergeCapabilitiesWithFields } from '../merge_capabilities_with_fields';
 import { getCapabilitiesForRollupIndices } from '../map_capabilities';
 
-const ROLLUP_INDEX_CAPABILITIES_METHOD = 'rollup.rollupIndexCapabilities';
-
-const getRollupIndices = (rollupData: { [key: string]: any[] }) => Object.keys(rollupData);
+const getRollupIndices = (rollupData: { [key: string]: any }) => Object.keys(rollupData);
 
 const isIndexPatternContainsWildcard = (indexPattern: string) => indexPattern.includes('*');
 const isIndexPatternValid = (indexPattern: string) =>
   indexPattern && isString(indexPattern) && !isIndexPatternContainsWildcard(indexPattern);
 
 export const getRollupSearchStrategy = (
   AbstractSearchStrategy: any,
-  RollupSearchRequest: any,
   RollupSearchCapabilities: any,
-  callWithRequestFactory: CallWithRequestFactoryShim
+  getRollupService: any
```

Comment:
I think we can avoid `any` here with `(reg: ReqFacade) => ILegacyScopedClusterClient`. Can you check it, please?

Reply:
I checked - (reg: ReqFacade) => Promise<ILegacyScopedClusterClient> is suitable for this case. I replaced any to this.

Reply:
Thank you!